test_generate_file_verbose_template_syntax_error fails with ./ difference in path

* Cookiecutter version: 2.0.2
* Template project url: none
* Python version: 3.9
* Operating System: openSUSE Tumbleweed

### Description:

Running the tests in the openSUSE build environment yields an error in the test function `test_generate_file_verbose_template_syntax_error` which is just a mismatch between `tests/files/syntax_error.txt` and `./tests/files/syntax_error.txt`.

### What I've run:

```
pytest-3.9 -v tests
[...]
[   10s] =================================== FAILURES ===================================
[   10s] _______________ test_generate_file_verbose_template_syntax_error _______________
[   10s] 
[   10s] env = <cookiecutter.environment.StrictEnvironment object at 0x7fa6e5baf040>
[   10s] expected_msg = 'Missing end of comment tag\n  File "./tests/files/syntax_error.txt", line 1\n    I eat {{ syntax_error }} {# this comment is not closed}'
[   10s] 
[   10s]     def test_generate_file_verbose_template_syntax_error(env, expected_msg):
[   10s]         """Verify correct exception raised on syntax error in file before generation."""
[   10s]         with pytest.raises(TemplateSyntaxError) as exception:
[   10s]             generate.generate_file(
[   10s]                 project_dir=".",
[   10s]                 infile='tests/files/syntax_error.txt',
[   10s]                 context={'syntax_error': 'syntax_error'},
[   10s]                 env=env,
[   10s]             )
[   10s] >       assert str(exception.value) == expected_msg
[   10s] E       assert 'Missing end ...s not closed}' == 'Missing end ...s not closed}'
[   10s] E           Missing end of comment tag
[   10s] E         -   File "./tests/files/syntax_error.txt", line 1
[   10s] E         ?         --
[   10s] E         +   File "tests/files/syntax_error.txt", line 1
[   10s] E               I eat {{ syntax_error }} {# this comment is not closed}
[   10s] 
[   10s] tests/test_generate_file.py:136: AssertionError
[   10s] ------------------------------ Captured log call -------------------------------
[   10s] DEBUG    cookiecutter.generate:generate.py:144 Processing file tests/files/syntax_error.txt
[   10s] DEBUG    cookiecutter.generate:generate.py:159 Created file at ./tests/files/syntax_error.txt
[   10s] DEBUG    cookiecutter.generate:generate.py:162 Check tests/files/syntax_error.txt to see if it's a binary
```

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -11,7 +11,7 @@ repos:
         files: \.rst$
         require_serial: true
   - repo: https://github.com/psf/black.git
-    rev: 21.10b0
+    rev: 22.3.0
     hooks:
       - id: black
         language_version: python3

--- a/tests/test_generate_file.py
+++ b/tests/test_generate_file.py
@@ -133,7 +133,7 @@ def test_generate_file_verbose_template_syntax_error(env, expected_msg_regex):
             context={'syntax_error': 'syntax_error'},
             env=env,
         )
-    assert expected_msg_regex.match(str(exception.value)) is not None
+    assert expected_msg_regex.match(str(exception.value))
 
 
 def test_generate_file_does_not_translate_lf_newlines_to_crlf(env, tmp_path):

--- a/tests/test_generate_file.py
+++ b/tests/test_generate_file.py
@@ -1,6 +1,7 @@
 """Tests for `generate_file` function, part of `generate_files` function workflow."""
 import json
 import os
+import re
 
 import pytest
 from jinja2 import FileSystemLoader
@@ -114,17 +115,16 @@ def test_generate_file_with_false_condition(env):
 
 
 @pytest.fixture
-def expected_msg():
+def expected_msg_regex():
     """Fixture. Used to ensure that exception generated text contain full data."""
-    msg = (
+    return re.compile(
         'Missing end of comment tag\n'
-        '  File "./tests/files/syntax_error.txt", line 1\n'
-        '    I eat {{ syntax_error }} {# this comment is not closed}'
+        ' {2}File "(.[/\\\\])*tests[/\\\\]files[/\\\\]syntax_error.txt", line 1\n'
+        ' {4}I eat {{ syntax_error }} {# this comment is not closed}'
     )
-    return msg.replace("/", os.sep)
 
 
-def test_generate_file_verbose_template_syntax_error(env, expected_msg):
+def test_generate_file_verbose_template_syntax_error(env, expected_msg_regex):
     """Verify correct exception raised on syntax error in file before generation."""
     with pytest.raises(TemplateSyntaxError) as exception:
         generate.generate_file(
@@ -133,7 +133,7 @@ def test_generate_file_verbose_template_syntax_error(env, expected_msg):
             context={'syntax_error': 'syntax_error'},
             env=env,
         )
-    assert str(exception.value) == expected_msg
+    assert expected_msg_regex.match(str(exception.value)) is not None
 
 
 def test_generate_file_does_not_translate_lf_newlines_to_crlf(env, tmp_path):

